prefect_kubernetes/jobs.py

```diff
@@ -0,0 +1,301 @@
+        kubernetes_credentials: `KubernetesCredentials` block
+            holding authentication needed to generate the required API client.
+        **kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+
+    Returns:
+        A Kubernetes `V1Job` object.
+
+    Example:
+        Create a job in the default namespace:
+        ```python
+        from prefect import flow
+        from prefect_kubernetes.credentials import KubernetesCredentials
+        from prefect_kubernetes.jobs import create_namespaced_job
+        from kubernetes.client.models import V1Job
+
+        @flow
+        def kubernetes_orchestrator():
+            v1_job_metadata = create_namespaced_job(
+                body=V1Job(**{"metadata": {"name": "test-job"}}),
+                kubernetes_credentials=KubernetesCredentials.load("k8s-creds"),
+            )
+        ```
+    """
+    api_client = kubernetes_credentials.get_batch_client()
+
+    return await run_sync_in_worker_thread(
+        api_client.create_namespaced_job, namespace=namespace, body=body, **kube_kwargs
+    )
+
+
+@task
+async def delete_namespaced_job(
+    job_name: str,
+    kubernetes_credentials: KubernetesCredentials,
+    body: Optional[V1DeleteOptions] = None,
+    namespace: Optional[str] = "default",
+    **kube_kwargs: Optional[Dict[str, Any]],
+) -> V1Status:
+    """Task for deleting a namespaced Kubernetes job.
+
+    Args:
+        job_name: The name of a job to delete.
+        kubernetes_credentials: `KubernetesCredentials` block
+            holding authentication needed to generate the required API client.
+        body: A Kubernetes `V1DeleteOptions` object.
+        namespace: The Kubernetes namespace to delete this job in.
+        **kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+
+    Returns:
+        A Kubernetes `V1Status` object.
+
+    Example:
+        Delete "my-job" in the default namespace:
+        ```python
+        from kubernetes.client.models import V1DeleteOptions
+        from prefect import flow
+        from prefect_kubernetes.credentials import KubernetesCredentials
+        from prefect_kubernetes.jobs import delete_namespaced_job
+
+        @flow
+        def kubernetes_orchestrator():
+            v1_job_status = delete_namespaced_job(
+                job_name="my-job",
+                kubernetes_credentials=KubernetesCredentials.load("k8s-creds"),
+                body=V1DeleteOptions(propagation_policy="Foreground"),
+            )
+        ```
+    """
+
+    api_client = kubernetes_credentials.get_batch_client()
+
+    if body:
+        kube_kwargs.update(body=V1DeleteOptions(**body))
+
+    return await run_sync_in_worker_thread(
+        api_client.delete_namespaced_job,
+        name=job_name,
+        namespace=namespace,
+        **kube_kwargs,
+    )
```

Comment:
Sorry, I might have been looking at an outdated version when I was reviewing.

I think this can be:
```
    return await run_sync_in_worker_thread(
        api_client.delete_namespaced_job,
        name=job_name,
        namespace=namespace,
        body=body,
        **kube_kwargs,
    )
```